support/cas-server-support-x509-core/src/main/java/org/apereo/cas/adaptors/x509/authentication/principal/AbstractX509PrincipalResolver.java

```diff
@@ -143,6 +143,14 @@ protected String getAlternatePrincipal(final X509Certificate certificate) {
             } catch (final CertificateParsingException e) {
                 LOGGER.warn("Error parsing subject alternative names to get rfc822 email [{}]", e.getMessage());
             }
+            try {
+                val x509subjectUPN = X509UPNExtractor.extractUPNString(certificate);
+                if (x509subjectUPN != null) {
+                    attributes.put("x509subjectUPN", CollectionUtils.wrapList(x509subjectUPN));
+                }
+            } catch (final CertificateParsingException e) {
+                LOGGER.warn("Error parsing subject alternative names to get User Principal Name as an attribute [{}]", e.getMessage());
```

Comment:
log the stack trace please, perhaps under debug or trace levels

Reply:
I did the exception conditionally inside the log statement. It seems to work although Intellij claims there are more arguments than parameters, but you either get a stacktrace or not depending on the log level. Let me know if you don't like it.